src/fields/field2_128/constants.rs

```diff
@@ -44,17 +43,67 @@ pub(crate) const fn subfield_basis() -> [Field2_128; Field2_128::SUBFIELD_BIT_LE
     ]
 }
 
+/// The precomputed lower-upper decomposition of the basis of the [`Field2_128`] subfield
+/// ([`subfield_basis`]). Used to attempt projection of GF(2^128) elements into the subfield, as
+/// described in [FS24 section 3.3][1].
+///
+/// [1]: https://eprint.iacr.org/2024/2010.pdf#subsection.3.3
+pub(crate) const fn subfield_basis_lu_decomposition() -> SubfieldBasisLowerUpperDecomposition {
+    // Computed using fields::field2_128::constants::test::compute_subfield_basis_lu_decomposition.
+    // We can't evaluate that function in const because its loops call methods on trait `Iterator`
+    // and related types.
+    SubfieldBasisLowerUpperDecomposition {
+        upper: [
```

Comment:
I put these constants in binary because while we represent things as u128 or u16 so we can efficiently XOR values together, conceptually these are matrices of bits. I think this layout of the constants helps to visualize how this works.